packages/real-time/test/unit/mutation-event-sender.spec.ts

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable @typescript-eslint/unbound-method */
+import sinon from 'sinon';
+import { ServerResponse } from 'http';
+import type { MutantResult } from 'mutation-testing-report-schema';
+
+import { MutationEventSender } from '../../src/mutation-event-sender.js';
+import { expect } from 'chai';
+
+describe(MutationEventSender.name, () => {
+  let responseMock: sinon.SinonStubbedInstance<ServerResponse>;
+  let onCompleteStub: sinon.SinonStub<[], void>;
+
+  beforeEach(() => {
+    responseMock = sinon.createStubInstance(ServerResponse);
```

Comment:
It seems a bit strange to me to make a stub instance for something like a http response. Is it not possible to make the object ourselves and spy on it? Especially since the sut is doing quite a lot of side-effect calling on the response this doesn't give me faith that it is being used properly

Reply:
It is a unit test. I want to avoid creating an _actual_ response. 

Do you mean to say it's better to write a custom fake that implements the `ServerResponse` interface?

Reply:
Well, I expected `ServerResponse` to just be a data object carrying some stream objects or something. That way you could test the streams. If that's not the case, this might be the simplest approach. Then a custom fake doesn't have much advantage over a stub

Reply:
I clearly expected too much 🙃

Reply:
Haven't you learned by now? This is JavaScript after all 😉